ark/utils/test_utils.py

```diff
@@ -0,0 +1,256 @@
+
+    right = (img_shape[1] - 1) // 2
+    left = (img_shape[1] + 2) // 2
+    up = (img_shape[0] - 1) // 2
+    down = (img_shape[0] + 2) // 2
+
+    counter = 1
+    for fov in range(fov_number):
+        for comp in range(comp_number):
+            label_data[fov, :up, :right, comp] = counter
+            counter = (counter % 255) + 1
+            label_data[fov, :up, left:, comp] = counter
+            counter = (counter % 255) + 1
+            label_data[fov, down:, :right, comp] = counter
+            counter = (counter % 255) + 1
+            label_data[fov, down:, left:, comp] = counter
+            counter = (counter % 255) + 1
+
+    return label_data
+
+
+def _create_tifs(base_dir, fov_names, img_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(img_names), shape, fills, dtype)
+
+    if sub_dir is None:
+        sub_dir = ""
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        filelocs[fov] = []
+        fov_path = os.path.join(base_dir, fov, sub_dir)
+        os.makedirs(fov_path)
+        for j, name in enumerate(img_names):
+            io.imsave(os.path.join(fov_path, f'{name}.tiff'), tif_data[i, :, :, j])
+            filelocs[fov].append(os.path.join(fov_path, name))
+
+    return filelocs, tif_data
+
+
+def _create_multitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        io.imsave(tiffpath, tif_data[i, :, :, :], plugin='tifffile')
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+def _create_mibitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    mass_map = tuple(enumerate(channel_names, 1))
+
+    for i, fov in enumerate(fov_names):
+        tif_obj = mi.MibiImage(tif_data[i, :, :, :],
+                               mass_map,
+                               **MIBITIFF_METADATA)
+
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        tiff.write(tiffpath, tif_obj, dtype=dtype)
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+def _create_reverse_multitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(channel_names), len(fov_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        io.imsave(tiffpath, tif_data[:, :, :, i], plugin='tifffile')
+        filelocs[fov] = tiffpath
+
+    tif_data = np.swapaxes(tif_data, 0, -1)
+
+    return filelocs, tif_data
+
+
+def _create_labels(base_dir, fov_names, comp_names, shape, sub_dir, fills, dtype):
```

Comment:
I understand these _helper_functions won't be directly called by anyone when writing tests, but it could be helpful for us to indicate in the name which ones are creating the data and which are saving it as a way to separate out the similarly named ones

Reply:
Sounds good.  btw, how are we going to handle internal functions in the docs?  They should definitely have doc strings, but it would maybe be useful to deliberately ignore adding functions prefixed with '_' to the docs (maybe a setting in sphinx?)

Reply:
Yeah, maybe @alex-l-kong can look into that. I'm certain that it's possible to ignore specific files, so worst comes to worst we can do that.

Reply:
apparently, sphinx already ignores `_` prefixed functions (see doc build on this PR for reference).